framework_tvb/tvb/interfaces/web/controllers/simulator/simulator_controller.py

```diff
@@ -341,12 +342,18 @@ def set_integrator(self, **data):
 
     @expose_fragment('simulator_fragment')
     def set_integrator_params(self, **data):
-        session_stored_simulator, is_simulation_copy, is_simulation_load, _ = self.context.get_common_params()
+        session_stored_simulator, is_simulation_copy, is_simulation_load, is_branch = self.context.get_common_params()
 
         if cherrypy.request.method == POST_REQUEST:
-            form = get_form_for_integrator(type(session_stored_simulator.integrator))()
+            form = get_form_for_integrator(type(session_stored_simulator.integrator))(is_branch)
```

Comment:
How about instead of the try-except block, if is_branch is True, we add the dt with session_stored_simulator.integrator.dt value in the data dict before calling fill_from_post?

Reply:
Yeah, that should work too.